node/pkg/utils/goroutine-pool/pool.go

```diff
@@ -0,0 +1,40 @@
+package goroutine_pool
+
+import (
+	"context"
+)
+
+type PoolJob func()
+
+type Pool struct {
+	jobChannel chan PoolJob
+}
+
+const POOL_WORKER_COUNT = 3
```

Comment:
let's make it customizable. it'd be good if this can be passed through `NewPool()`
let's include into `Pool` struct